Assets/Plugins/Editor/Uplift/UpdateTool/Updater.cs

```diff
@@ -0,0 +1,339 @@
+m4skA/WsUsMCfsZY27RiZ+B162+laDD0oRiovOx5zKrzG/3yAFVutz2Bfud3QenU
+3zt14ttnzFCcMlglW9oqffHMwQ1Smx/30ccDonoa1E02hnNmxeLyBwomPqd/ZYy6
+zSJW0aSi1DadkZsifpr65AwgSuN5uGEhQas0glsN
+-----END CERTIFICATE-----
+"
+        };
+        private static bool trustUnknown;
+
+        public static void UpdateUplift(string url)
+        {
+            trustUnknown = UpliftPreferences.TrustUnknownCertificates();
+            string unitypackageName = url.Split('/').Last();
+            DirectoryInfo assetsPath = new DirectoryInfo(Application.dataPath);
+            string destination = Path.Combine(assetsPath.Parent.FullName, unitypackageName);
+            
+            ServicePointManager.ServerCertificateValidationCallback = CertificateValidationCallback;
+            using (var client = new WebClient())
+            {
+                client.DownloadFile(url, destination);
+            }
+
+            AssetDatabase.ImportPackage(destination, false);
+        }
+
+        public static void CheckForUpdate(bool forceCheck = false)
+        {
+            if(forceCheck || ShouldCheck())
+            {
+                EditorApplication.update += EditorUpdate;
+    			updateCoroutine = CheckForUpdateRoutine();
+            }
+        }
+        
+        private static void EditorUpdate()
+        {
+            updateCoroutine.MoveNext();
+        }
+
+        private static bool ShouldCheck()
+        {
+            string lastCheck = EditorPrefs.GetString(lastUpdateCheckKey);
+            if(string.IsNullOrEmpty(lastCheck)) return true;
+            
+            try
+            {
+                TimeSpan timeFromLastCheck = DateTime.UtcNow - DateTime.ParseExact(lastCheck, dateFormat, provider);
+                return timeFromLastCheck > TimeSpan.FromDays(1);
+            }
+            catch(FormatException e)
+            {
+                Debug.LogErrorFormat("The date stored for the last check of Uplift ({0}) is not in the correct format:\n{1}", lastCheck, e);
+            }
+            return true;
+        }
+        
+        private static IEnumerator CheckForUpdateRoutine() {
+            IEnumerator e = GetReleasesJson();
+            while (e.MoveNext()) yield return e.Current;
+
+            string json = (string) e.Current;
+
+            if (json == null) {
+                Debug.LogError ("Unable to check for Uplift updates");
+            } else {
+                System.Object obj = MiniJSON.Json.Deserialize (json);
+                List<System.Object> releases = (List<System.Object>)obj;
+                foreach (Dictionary<string, object> release in releases)
+                {
+                    if(VersionParser.GreaterThan((string)release ["tag_name"], About.Version))
+                    {
+                        Debug.Log("There is a new version of Uplift available!");
+                        UpdatePopup popup = EditorWindow.GetWindow(typeof(UpdatePopup), true) as UpdatePopup;
```

Comment:
I am not sure how that works if we have had multiple updates, e.g. upgrade from 1.0.0beta1 to 1.0.0beta4.